index.html

```diff
@@ -1 +1,38 @@
-Hello 
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>jaylol</title>
+    <link rel="stylesheet" href="./style.css">
+    <link rel="icon" href="./favicon.ico" type="image/x-icon">
+  </head>
+  <body>
+    <main>
+        <h1>jaylol</h1>  
+    </main>
+    <script src="https://accounts.google.com/gsi/client" onload="console.log('TODO: add onload function')">  
+    </script>
+    <script>
+        const code_receiver_uri = 'https://jaylol.com/code_receiver';
+        const client = google.accounts.oauth2.initCodeClient({
+        client_id: '117883256410-cj6gkhqi8ih2gs3245jpteo6cb8u8qvl.apps.googleusercontent.com',
+        scope: 'https://www.googleapis.com/auth/userinfo.email',
+        ux_mode: 'popup',
+        callback: (response) => {
+            const xhr = new XMLHttpRequest();
+            xhr.open('POST', code_receiver_uri, true);
+            xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+            // Set custom header for CRSF
+            xhr.setRequestHeader('X-Requested-With', 'XmlHttpRequest');
+            xhr.onload = function() {
+            console.log('Auth code response: ' + xhr.responseText);
+            };
+            xhr.send('code=' + response.code);
+        },
+        });
+    </script>
+    <button onclick="client.requestCode();">Authorize with Google</button>
```

Comment:
this is the only block of code in the file that is of-interest for figuring out how to set up google sign in via oauth 2.0 authorization code flow